src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -1147,14 +1147,16 @@ private Task WriteStringAsync(string s)
             if (s.Length <= _writeBuffer.Length - offset)
             {
                 byte[] writeBuffer = _writeBuffer;
+                int mask = StaticHttpSettings.EncodingValidationMask;
                 foreach (char c in s)
```

Comment:
Should we change `c` to `int` as in other cases?

Reply:
I did not like the implicit casting happening in the `foreach` expression. If anything. would prefer rather `((int)c & mask) != 0` to make more explicit.

Reply:
But this is no strong opinion, may change to `foreach (int c in s)` as well.

Reply:
In the end decided to change for consistency 😄